src/requests/is-verified-request.spec.ts

```diff
@@ -0,0 +1,239 @@
+import * as assert from 'assert'
+
+import { isVerifiedRequest } from './is-verified-request'
+import { CanonicalRequest, Secret } from './typings'
+import { createSignature } from './create-signature'
+import { ContentfulSigningHeader } from './constants'
+
+const makeRequest = ({
+  path = '/api/v1/resources/1',
+  method = 'GET',
+  headers,
+  signedHeaders,
+  body,
+}: {
+  path?: string
+  method?: 'GET' | 'POST'
+  headers?: Record<string, string>
+  signedHeaders?: string[]
+  body?: string
+}): CanonicalRequest => {
+  return {
+    method,
+    path,
+    headers,
+    signedHeaders,
+    body,
+  }
+}
+
+const makeIncomingRequest = (request: CanonicalRequest) => {
+  const now = Date.now()
+  const signature = createSignature(VALID_SECRET, request, now)
+
+  const headers = {
+    ...request.headers,
+    [ContentfulSigningHeader.Signature]: signature,
+    [ContentfulSigningHeader.Timestamp]: now.toString(),
+  }
+
+  if (request.signedHeaders) {
+    headers[ContentfulSigningHeader.SignedHeaders] = request.signedHeaders.join(',')
+  }
+
+  return { ...request, headers }
+}
+
+const VALID_SECRET: Secret = new Array(64).fill('a').join('')
+
+describe('isVerifiedRequest', () => {
+  it('verifies a verified request', () => {
```

Comment:
Maybe it's worth adding a test case to check that verification fails if you use a different secret?

Reply:
Good catch. It's there for the other method, forgot for this. Thanks 👍